ml-agents-envs/mlagents/envs/environment.py

```diff
@@ -494,7 +495,7 @@ def _get_state(self, output: UnityRLOutputProto) -> AllBrainInfo:
 
     @staticmethod
     def _parse_side_channel_message(
-        side_channels: Dict[int, SideChannel], data: bytearray
+        side_channels: Dict[int, SideChannel], data: bytes
```

Comment:
`bytes` is immutable - these seemed better for processing inputs. I left outputs as` bytearray`s though.